valhalla/midgard/gridded_data.h

```diff
@@ -100,14 +116,325 @@ template <class coord_t> class GriddedData : public Tiles<coord_t> {
    *
    * @return contour line geometries with the larger intervals first (for rendering purposes)
    */
-  contours_t GenerateContours(const std::vector<float>& contour_intervals,
+  contours_t GenerateContours(std::vector<contour_interval_t>& intervals,
                               const bool rings_only = false,
                               const float denoise = 1.f,
-                              const float generalize = 200.f) const;
+                              const float generalize = 200.f) const {
+    // sort the contours first on the metric index then on the values with the bigger contours first
```

Comment:
this used to have to be done outside of this class and so was kind of like an unspoken rule. now its part of the contract of this method to have sorted output